clustered/common/src/main/java/org/ehcache/clustered/common/messages/EhcacheCodec.java

```diff
@@ -34,30 +34,28 @@
 
   @Override
   public byte[] encodeMessage(EhcacheEntityMessage message) {
-    if (message.getType() == EhcacheEntityMessage.Type.LIFECYCLE_OP) {
-      return LifeCycleOpCodec.encode((LifecycleMessage) message);
-    } else {
-      return ServerStoreOpCodec.encode((ServerStoreOpMessage) message);
-    }
+    return message.encode();
```

Comment:
I do not get that change. Why did you decided to have messages carry their encoding / decoding logic?

I understand I gave that advice for the work @albinsuresh is doing, but in my mind the context here is different.